Gems/Terrain/Code/Source/EditorComponents/EditorTerrainPhysicsColliderComponent.cpp

```diff
@@ -51,7 +52,7 @@ namespace Terrain
                     ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                     ->DataElement(AZ::Edit::UIHandlers::Default, &TerrainPhysicsColliderConfig::m_defaultMaterialAsset,
                         "Default Surface Physics Material", "Select a material to be used by unmapped surfaces by default")
-                        ->Attribute(AZ::Edit::Attributes::DefaultAsset, &TerrainPhysicsColliderConfig::GetDefaultPhysicsAssetId)
+                        ->Attribute(AZ::Edit::Attributes::DefaultAsset, &TerrainPhysicsColliderConfig::GetDefaultPhysicsMaterialAssetIdEditContext)
```

Comment:
update: Actually looking at this you shouldn't need to add the member function at all, you can just move `static AZ::Data::AssetId GetDefaultPhysicsMaterialAssetId()` up and then pass it here....

```
->Attribute(AZ::Edit::Attributes::DefaultAsset, &GetDefaultPhysicsMaterialAssetId)
```

I'm pretty sure that should work no?

Reply:
I'm pretty sure free functions do work, so you can remove the member functions and friends entirely and keep the function in the .cpp

Reply:
Nice! It works! I'll remove the function members. I thought this line was checking the passed type belonged to the class https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/EditContext.h#L903

Thanks!

Reply:
Changed. Thanks!